lib/p5.sound.js

```diff
@@ -6771,11 +6769,20 @@ effect = function () {
     this.output.disconnect();
     this.output = undefined;
   };
+  p5.Effect.prototype.dispose = function () {
+    var index = p5sound.soundArray.indexOf(this);
+    p5sound.soundArray.splice(index, 1);
+    this.input.disconnect();
+    this.input = undefined;
+    this.output.disconnect();
+    this.output = undefined;
```

Comment:
ah, thanks @GoToLoop ! Sounds like we should use `delete` for all dispose methods